src/libraries/System.IO.FileSystem/tests/FileStream/FileStreamConformanceTests.Windows.cs

```diff
@@ -176,4 +176,148 @@ public struct SHARE_INFO_502
+                Assert.False(fileStream.CanSeek); // #54143
+
+                try
+                {
+                    CancellationTokenSource cts = new(TimeSpan.FromMilliseconds(250));
+
+                    await fileStream.ReadAsync(new byte[4096], cts.Token);
+                }
+                catch (OperationCanceledException)
+                {
+                    // most likely there is no data available and the task is going to get cancelled
+                    // which is fine, we just want to make sure that reading from devices is supported (#54143)
+                }
+            }
+        }
+
+        private static FileStream? OpenFirstAvailableDeviceInterface()
+        {
+            const int DIGCF_PRESENT = 0x2;
+            const int DIGCF_DEVICEINTERFACE = 0x10;
+            const int ERROR_NO_MORE_ITEMS = 259;
+
+            HidD_GetHidGuid(out Guid HidGuid);
+            IntPtr deviceInfoSet = SetupDiGetClassDevs(in HidGuid, IntPtr.Zero, IntPtr.Zero, DIGCF_PRESENT | DIGCF_DEVICEINTERFACE);
+
+            try
+            {
+                SP_DEVINFO_DATA deviceInfoData = new SP_DEVINFO_DATA();
+                deviceInfoData.cbSize = (uint)Marshal.SizeOf(deviceInfoData);
+
+                uint deviceIndex = 0;
+                while (SetupDiEnumDeviceInfo(deviceInfoSet, deviceIndex++, ref deviceInfoData))
+                {
+                    if (Marshal.GetLastWin32Error() == ERROR_NO_MORE_ITEMS)
+                    {
+                        break;
+                    }
+
+                    SP_DEVICE_INTERFACE_DATA deviceInterfaceData = new SP_DEVICE_INTERFACE_DATA();
+                    deviceInterfaceData.cbSize = Marshal.SizeOf(deviceInterfaceData);
+
+                    if (!SetupDiEnumDeviceInterfaces(deviceInfoSet, IntPtr.Zero, in HidGuid, deviceIndex, ref deviceInterfaceData))
+                    {
+                        continue;
+                    }
+
+                    SP_DEVICE_INTERFACE_DETAIL_DATA deviceInterfaceDetailData = new SP_DEVICE_INTERFACE_DETAIL_DATA();
+                    deviceInterfaceDetailData.cbSize = IntPtr.Size == 8 ? 8 : 6;
+
+                    uint size = (uint)Marshal.SizeOf(deviceInterfaceDetailData);
+
+                    if (!SetupDiGetDeviceInterfaceDetail(deviceInfoSet, ref deviceInterfaceData, ref deviceInterfaceDetailData, size, ref size, IntPtr.Zero))
+                    {
+                        continue;
+                    }
+
+                    string devicePath = deviceInterfaceDetailData.DevicePath;
+                    Assert.StartsWith(@"\\?\hid", devicePath);
+
+                    try
+                    {
+                        return new FileStream(devicePath, FileMode.Open, FileAccess.Read, FileShare.Read, 0, FileOptions.Asynchronous);
+                    }
+                    catch (IOException)
+                    {
+                        continue; // device has been locked by another process
```

Comment:
Is there a more precise way to identify "device has been locked by another process"? I'm asking because the exception reported in https://github.com/dotnet/runtime/issues/54143 is of type `IOException` as well.

Reply:
In #54143 the exception was thrown for `FileStream.ReadAsync`. 

I could most probably call `GetLastError` here and check for exact exception, but I don't think that it's worth it as most probably we would find other edge cases why device can't be opened very soon and would have to fix the test again.